base/c.jl

```diff
@@ -140,7 +140,7 @@ NULs are handled like any other character (i.e. the output will be NUL-terminate
 only if the `src` is).
 """
 function transcode end
-transcode{T<:Union{UInt8,UInt16}}(::Type{T}, src::Vector{T}) = src
+transcode{T<:Union{UInt8,UInt16,UInt32}}(::Type{T}, src::Vector{T}) = src
```

Comment:
This is not used for the fix, but I think it is reasonable for not Windows systems